tests/test_deobfuscate_filenames.py

```diff
@@ -54,29 +60,83 @@ def test_is_probably_obfuscated(self):
 
         # Create directory (with a random directory name)
         dirname = os.path.join(SAB_DATA_DIR, "testdir" + str(random.randint(10000, 99999)))
         os.mkdir(dirname)
 
         # Create a big enough file with a non-useful filename
-        output_file = dirname + "/599c1c9e2bdfb5114044bf25152b7eaa.mkv"
-        with open(output_file, "wb") as myfile:
-            # must be above MIN_SIZE, so ... 15MB
-            myfile.truncate(15 * 1024 * 1024)
-        # Check it exists now:
-        assert os.path.isfile(output_file)
+        output_file1 = os.path.join(dirname, "111c1c9e2bdfb5114044bf25152b7eaa.bla")
+        create_big_file(output_file1)
+        assert os.path.isfile(output_file1)
+
+        # ... and another one
+        output_file2 = os.path.join(dirname, "333c1c9e2bdfb5114044bf25152b7eaa.bla")
+        create_big_file(output_file2)
+        assert os.path.isfile(output_file2)
+
+        # and a useful filename ... so that should not get renamed
+        output_file3 = os.path.join(dirname, "This Great Download 2020.bla")
+        create_big_file(output_file3)
+        assert os.path.isfile(output_file3)
 
         # and now unleash the magic on that directory, with a more useful jobname:
         jobname = "My Important Download 2020"
-        deobfuscate(dirname, jobname)
-        # Check if file was renamed
-        assert not os.path.exists(output_file)  # original filename should not be there anymore
-        assert os.path.exists(os.path.join(dirname, jobname + ".mkv"))  # ... it should be renamed to the jobname
+        deobfuscate_dir(dirname, jobname)
+
+        # Check original files:
+        assert not os.path.exists(output_file1)  # original filename should not be there anymore
+        assert not os.path.exists(output_file2)  # original filename should not be there anymore
+        assert os.path.exists(output_file3)  # original filename should still be there
+
+        # Check the renaming
+        assert os.path.exists(os.path.join(dirname, jobname + ".bla"))  # ... it should be renamed to the jobname
+        assert os.path.exists(os.path.join(dirname, jobname + ".1.bla"))  # ... it should be renamed to the jobname
+
+        # Done. Remove (non-empty) directory
+        shutil.rmtree(dirname)
+
+    def test_deobfuscate_filelist(self):
+        # Full test: a filelist with a non-useful named file in it: Test that deobfuscate() works and renames it
+        # ... but not the files that are NOT in the filelist (although in same directory)
+
+        # Create directory (with a random directory name)
+        dirname = os.path.join(SAB_DATA_DIR, "testdir" + str(random.randint(10000, 99999)))
+        os.mkdir(dirname)
+
+        # Create a big enough file with a non-useful filename
+        output_file1 = os.path.join(dirname, "111c1c9e2bdfb5114044bf25152b7eaa.bla")
+        create_big_file(output_file1)
```

Comment:
Is this test different than the one above? Adds to my feeling we can do with just 1 function and leave the directory listing out :)

Reply:
Already removie in the commit deobfuscate: removed deobfuscate_dir()